packages/contracts-bedrock/contracts/L1/SystemConfig.sol

```diff
@@ -207,4 +221,18 @@ contract SystemConfig is OwnableUpgradeable, Semver {
             sstore(slot, _unsafeBlockSigner)
         }
     }
+
+    /**
+     * @notice Returns the minimum gas limit allowed by the system. If the L2
+     *         gas limit is set to a value smaller than this, then it is
+     *         possible for a block to be produced that uses more gas than what
+     *         is allowed on L2, resulting in a liveness failure. The MAX_RESOURCE_LIMIT
+     *         represents the amount of gas that can be consumed by deposits and
+     *         the SYSTEM_TRANSACTION_MAX_GAS represents the maximum amount of
+     *         gas the system transaction can consume in a block.
+     */
+    function minimumGasLimit() public view returns (uint256) {
+        uint256 maxResourceLimit = uint256(ResourceMetering(PORTAL).MAX_RESOURCE_LIMIT());
```

Comment:
I find this relationship pretty ugly. If anything, `ResourceMetering` should be pulling the max resource limit from `SystemConfig` rather than the other way around. That way you don't need a minimum gas limit like this. You just derive the maximum available deposit gas from the maximum resource gas as defined in this contract.

Reply:
Considered this design but decided against it balancing amount of diff

Reply:
Also seems like a cleaner layout to me, but this feels like a change that should be done after we ship Bedrock.